fe/fe-core/src/test/java/com/starrocks/credential/AWSCloudConfigurationTest.java

```diff
@@ -17,27 +17,53 @@
 import com.staros.proto.FileStoreInfo;
 import com.starrocks.credential.aws.AWSCloudConfiguration;
 import com.starrocks.credential.aws.AWSCloudCredential;
+import com.starrocks.credential.provider.OverwriteAwsDefaultCredentialsProvider;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.s3a.AWSCredentialProviderList;
+import org.apache.hadoop.fs.s3a.S3AFileSystem;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.junit.Assert;
 import org.junit.Test;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
 
+import java.net.URI;
 import java.util.HashMap;
 import java.util.Map;
 
 public class AWSCloudConfigurationTest {
 
     @Test
-    public void testUseAWSSDKDefaultBehavior() {
+    public void testUseAWSSDKDefaultBehavior() throws Exception {
         // Test hadoop configuration
         Map<String, String>  properties = new HashMap<>();
         properties.put("aws.s3.use_aws_sdk_default_behavior", "true");
         CloudConfiguration cloudConfiguration = CloudConfigurationFactory.buildCloudConfigurationForStorage(properties);
         Assert.assertNotNull(cloudConfiguration);
         Configuration configuration = new Configuration();
         cloudConfiguration.applyToConfiguration(configuration);
-        Assert.assertEquals("software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider",
+        Assert.assertEquals(OverwriteAwsDefaultCredentialsProvider.class.getName(),
                 configuration.get("fs.s3a.aws.credentials.provider"));
+        S3AFileSystem fs = (S3AFileSystem) FileSystem.get(new URI("s3://hi/a.parquet"), configuration);
+        AWSCredentialProviderList list =  fs.shareCredentials("ut");
+        int previousHashCode = list.getProviders().get(0).hashCode();
```

Comment:
shall assert the previous and current provider are the same instance/class type, AKA the aws.DefaultCredentialsProvider